### PR TITLE
[Feature][Engine] Unify job env parameters

### LINKED ISSUE #5937: [Feature][Engine] Unify job env parameters
### Search before asking

- [X] I had searched in the [feature](https://github.com/apache/seatunnel/issues?q=is%3Aissue+label%3A%22Feature%22) and found no similar feature requirement.

### Description

[Engine] Unify job env parameters

### Usage Scenario

_No response_

### Related issues

_No response_

### Are you willing to submit a PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by happyboy1024**
### Purpose of this pull request

1. Adjust the flink parameter configuration mode by prefixing `flink.` and the parameter items provided by flink official
2. Clear outdated configurations in the md file and e2e
close #5937
### How was this patch tested?

add new test cases to separately verify that the parameters are consistent before and after adjustments

### Check list

* [ ] If any new Jar binary package adding in your PR, please add License Notice according
  [New License Guide](https://github.com/apache/seatunnel/blob/dev/docs/en/contribution/new-license.md)
* [ ] If necessary, please update the documentation to describe the new feature. https://github.com/apache/seatunnel/tree/dev/docs
* [ ] If you are contributing the connector code, please check that the following files are updated:
  1. Update change log that in connector document. For more details you can refer to [connector-v2](https://github.com/apache/seatunnel/tree/dev/docs/en/connector-v2)
  2. Update [plugin-mapping.properties](https://github.com/apache/seatunnel/blob/dev/plugin-mapping.properties) and add new connector information in it
  3. Update the pom file of [seatunnel-dist](https://github.com/apache/seatunnel/blob/dev/seatunnel-dist/pom.xml)
* [ ] Update the [`release-note`](https://github.com/apache/seatunnel/blob/dev/release-note.md).

### CODE REVIEW

**Comment by hailin0** [comment]
Please check ci

File: seatunnel-core/seatunnel-flink-starter/seatunnel-flink-starter-common/src/main/java/org/apache/seatunnel/core/starter/flink/utils/EnvironmentUtil.java [hunk 1]
```diff
@@ -119,15 +119,47 @@ public static void initConfiguration(Config config, Configuration configuration)
...
+            for (Map.Entry<String, ConfigValue> entryConfKey : config.entrySet()) {
+                String confKey = entryConfKey.getKey().trim();
                 if (confKey.startsWith(prefixConf)) {
                     configuration.setString(
-                            confKey.replaceFirst(prefixConf, ""), entryConfKey.getValue().render());
+                            confKey.replaceFirst(replacePrefix, ""),
+                            entryConfKey.getValue().unwrapped().toString());
                 }
             }
         }
```
Comment (Hisoka-X) [inline seatunnel-core/seatunnel-flink-starter/seatunnel-flink-starter-common/src/main/java/org/apache/seatunnel/core/starter/flink/utils/EnvironmentUtil.java:153]: please add test case for this. 

ignore below
~~Look like it remove prefix of `flink.table.exec.config` to `table.exec.config`. Not worked on `flink.pipeline.max-parallelism` according doc https://github.com/apache/seatunnel/pull/6003/files#diff-8f9b660879c6a25ff9d0f64619025ea7be9a72cd0152c4b896843424cab6abbcR44~~

**Review by Hisoka-X — APPROVED** [review]

File: seatunnel-core/seatunnel-flink-starter/seatunnel-flink-starter-common/src/main/java/org/apache/seatunnel/core/starter/flink/utils/EnvironmentUtil.java (hunk 1 above)
Comment (Hisoka-X) [inline seatunnel-core/seatunnel-flink-starter/seatunnel-flink-starter-common/src/main/java/org/apache/seatunnel/core/starter/flink/utils/EnvironmentUtil.java:153]: Overall LGTM, but please add a test case for `flink.table.exec.` feature.

**Review by Hisoka-X — CHANGES_REQUESTED** [review]

File: seatunnel-core/seatunnel-flink-starter/seatunnel-flink-starter-common/src/main/java/org/apache/seatunnel/core/starter/flink/utils/EnvironmentUtil.java (hunk 1 above)
Comment (happyboy1024) [reply seatunnel-core/seatunnel-flink-starter/seatunnel-flink-starter-common/src/main/java/org/apache/seatunnel/core/starter/flink/utils/EnvironmentUtil.java:153]: > Overall LGTM, but please add a test case for `flink.table.exec.` feature.

I've added a test case for this that uses table env parallelism to override global parallelism. PTAL.

File: seatunnel-e2e/seatunnel-engine-e2e/connector-seatunnel-e2e-base/src/test/java/org/apache/seatunnel/engine/e2e/UnifyEnvParameterIT.java [hunk 2]
```diff
@@ -0,0 +1,303 @@
...
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicReference;
+
+import static org.awaitility.Awaitility.await;
+
+@Slf4j
+@DisabledOnContainer(
+        value = {},
+        type = {EngineType.SEATUNNEL, EngineType.SPARK},
```
Comment (Hisoka-X) [inline seatunnel-e2e/seatunnel-engine-e2e/connector-seatunnel-e2e-base/src/test/java/org/apache/seatunnel/engine/e2e/UnifyEnvParameterIT.java:48]: LGTM, but I think we should create new place to store test only for flink in the future.

**Review by Hisoka-X — APPROVED** [review]

File: seatunnel-e2e/seatunnel-engine-e2e/connector-seatunnel-e2e-base/src/test/java/org/apache/seatunnel/engine/e2e/UnifyEnvParameterIT.java (hunk 2 above)
Comment (happyboy1024) [reply seatunnel-e2e/seatunnel-engine-e2e/connector-seatunnel-e2e-base/src/test/java/org/apache/seatunnel/engine/e2e/UnifyEnvParameterIT.java:48]: > LGTM, but I think we should create new place to store test only for flink in the future.

Agree with you, we can add a new module under `seatunnel-engine-e2e`, this module can be called `seatunnel-engine-flink-e2e` or `seatunnel-flink-engine-e2e`.

Comment (Hisoka-X) [reply seatunnel-e2e/seatunnel-engine-e2e/connector-seatunnel-e2e-base/src/test/java/org/apache/seatunnel/engine/e2e/UnifyEnvParameterIT.java:48]: Yep. Would you mind to do this after this PR be merged?

Comment (happyboy1024) [reply seatunnel-e2e/seatunnel-engine-e2e/connector-seatunnel-e2e-base/src/test/java/org/apache/seatunnel/engine/e2e/UnifyEnvParameterIT.java:48]: > Yep. Would you mind to do this after this PR be merged?

Of course not, after this pr merge, I will do this.

**Review by hailin0 — APPROVED** [review]

**Pull request merged**